lib/maxwell/middleware/header.ex

```diff
@@ -20,20 +20,17 @@ defmodule Maxwell.Middleware.Headers do
   alias Maxwell.Conn
 
   def init(headers) do
-    check_headers(headers)
-    Conn.new()
-    |> Conn.put_req_headers(headers)
-    |> Map.get(:req_headers)
+    check_headers!(headers)
+    headers
```

Comment:
I think there is already an implicit order requirement for the HeaderCase middleware (i.e. that it comes after any other middleware which modifies the headers). The risk here is that duplicates won't be caught if the cases don't match. Depending on what the adapter does with duplicates, that may or may not be a problem.